src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1370,6 +1371,15 @@ BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter
                     var memberName = containingMember.GetMemberCallerName();
                     defaultValue = new BoundLiteral(syntax, ConstantValue.Create(memberName), Compilation.GetSpecialType(SpecialType.System_String)) { WasCompilerGenerated = true };
                 }
+                else if (callerSourceLocation is object && getArgumentIndex(parameter.CallerArgumentExpressionParameterIndex, argsToParamsOpt) is int argumentIndex &&
+                    argumentIndex > -1 && argumentIndex < arguments.Length)
+                {
+                    // PROTOTYPE(caller-expr): Do we need to support VB?
+                    // PROTOTYPE(caller-expr): Do we need to add a warning/error for pre-C# 10 using this feature?
```

Comment:
>// PROTOTYPE(caller-expr): Do we need to add a warning/error for pre-C# 10 using this feature? [](start = 20, length = 94)

Yes, we need to report an error when the feature is used while targeting earlier versions of the language.

Reply:
@AlekseyTs Does that applies also on callsites? For example:

- AssemblyA defines the attribute on some parameter of a function

    - If AssemblyA is using an ealier language version => error

- Now, AssemblyB references AssemblyA and tries to call this method:

    - If AssemblyB is using an ealier language version => Allow or Error?

Reply:
> Does that applies also on callsites

I am not sure I see a difference. If we are about to supply default value based on the attribute here, I think we should check language version and complain.

---
In reply to: [600239518](https://github.com/dotnet/roslyn/pull/51952#discussion_r600239518) [](ancestors = 600239518)